src/utils/ServiceRequest.ts

```diff
@@ -40,25 +50,46 @@ export class ServiceRequest<T> {
     Logger.setLogLevelFromSettings(settings);
     Settings.logSettings(settings.settings);
 
-    Logger.info(`${name} Request: `, request);
-    this.query = queryParams(request);
+    // Set request ID
+    this.headers = request.headers;
+    const requestIdHeaderList = [
+      'x-ms-kms-request-id',
+      'x-ms-request-id',
+      'x-request-id',
+      'request-id',
+      'requestid',
+    ]
+    const requestIdFromHeader = requestIdHeaderList
+      .map((header) => this.headers ? this.headers[header] : undefined)
+      .find((header) => header !== undefined);
+    if (!requestIdFromHeader) {
+      this.requestId = Date.now().toString();
+      Logger.warn(`Request ID not provided. Using current timestamp as request ID: ${this.requestId}`, this.logContext);
```

Comment:
To avoid a lot of warnings, do B&A services need to change their requests?
Is it even possible to modify all the requests in B&A? As far as I remember we can't change requests for public key endpoints.

Reply:
After looking at  [CI · microsoft/privacy-sandbox-dev@9dc0c53](https://github.com/microsoft/privacy-sandbox-dev/actions/runs/11853703157), this warning looks informative to me because it shows you how the request ID was generaleted. So I think we can leave it